### ISSUE TITLE
Add support for YAAC

### ISSUE THREAD

**Issue by sysadminpower2019** [richonguzman/LoRa_APRS_iGate#139]
Hello,

here is another multiplatform client that would be great to get to work with the LORA module.
I tried to set it up like the other examples here but i can't get it to connect neither bia TCP/IP nor via Serial. 

https://www.ka2ddo.org/ka2ddo/YAACdocs/config_kissovertcp.html

**Comment by richonguzman** [comment]
there is Serial output and also KISS output (TNC2 encapsulated)

did both not work?

**Comment by sysadminpower2019** [comment]
yea i tried both options and nothing happens.

**Comment by sysadminpower2019** [comment]
![Screenshot from 2024-07-19 16-59-35](https://github.com/user-attachments/assets/bc6f8183-6f79-4062-9467-a214d7bd5169)
Here is a screenshot from the serial option

**Comment by richonguzman** [comment]
will try to write the YAAC dev to know what I should do.

**Comment by Mane76** [comment]
Could you please try two variants, for both the TNC Option must be switched on in the iGate. I assume you run LINUX on your computer.

1) KISS over TCP: Enter IP from iGate, Port 8001, enter Callsign, enable Transmit, disable TEMP1-1, enable WIDE1-1, disable Beacon, Protocol RAW AX25

2) Serial Option: Device name /tmp/tnc, Bauf 115200, enter Callsign, enable Transmit, disable TEMP1-1, enable WIDE1-1, disable Beacon, Protocol RAW AX25
Then go to a console and enter  socat PTY,link=/tmp/tnc TCP:ip_of_iGate:8001 &

**Comment by Mane76** [comment]
as 3rd option (but this is not your origin question), you can run XAstir as APRS Client with socat out of option 2:

![UqTQkW3ozxdaAUVU](https://github.com/user-attachments/assets/1dd4dc36-8975-4873-bc5d-9ccc13151688)

**Comment by sysadminpower2019** [comment]
it does seem that either the tcp/ip or the serial worked as i was receiving stations however when i try to send out any beacons nothing shows up on aprs.fi. The lora-aprs module does show up regularly sending beacons but nothing appears from YAAC.

Would the socat method then also theortically work for PAT?

**Comment by sysadminpower2019** [comment]
at the moment. I am not entirely sure. there are not that many APRS stations around me to test if i am receiving and on which interface that is. I will need to try again

**Comment by Mane76** [comment]
> it does seem that either the tcp/ip or the serial worked as i was receiving stations however when i try to send out any beacons nothing shows up on aprs.fi. The lora-aprs module does show up regularly sending beacons but nothing appears from YAAC.
> 
> Would the socat method then also theortically work for PAT?

Take care not to mix up things:
When the iGate runs in TNC mode you can access it via Port 8001. This makes sense if either

- the iGate is not connected to APRS-IS and you want to see the stations on the map of XAstir or YAAC
- if the is or is not connected to APRS-IS and you want to send and receive messages through the iGates SSID via XAstir
- if the iGate is connected to APRS-IS and you additionally want to see the output

In all cases the functionality of connection to aprsis, digipeating, beaconing, ... shall be done via the iGate from Ricardo configurable in the webUI

**Comment by Mane76** [comment]
> Would the socat method then also theortically work for PAT?

Basically yes, socat "emulates" a serial KISS interface to PAT and a TCP KISS interface to the iGate

**Comment by sysadminpower2019** [comment]
yes, the lora-aprs module is transmitting correctly to aprs.fi the basic hardware installation works without any issues.

yes the ideal would be to be able to connect to the APRS-IS network via the hardware module, and then use YAAC or similar to send beacons, receive stations, send messages etc.

At the moment I am simply not sure if the TCP/IP is working or the Serial Variant is working, if anything is really working at all. 
All I can say is that at one point some additional station data did appear on YAAC. However I am certainly not able to transmit anything into the APRS-IS network from YAAC or similar

**Comment by Mane76** [comment]
> yes, the lora-aprs module is transmitting correctly to aprs.fi the basic hardware installation works without any issues.
> 
ok
> yes the ideal would be to be able to connect to the APRS-IS network via the hardware module, and then use YAAC or similar to send beacons, receive stations, send messages etc.
> 
ok except the beaconing. This should be done through the iGates configuration

> At the moment I am simply not sure if the TCP/IP is working or the Serial Variant is working, if anything is really working at all. All I can say is that at one point some additional station data did appear on YAAC. However I am certainly not able to transmit anything into the APRS-IS network from YAAC or similar

Socat and XAstir are working perfect with the configs I sent you, I have this running in test-setup

**Comment by sysadminpower2019** [comment]
In terms of sending a beacon from YAAC. Would it not be possible to set the LORA board with callsign:5 and then send a beacon from YAAC with callsign:15

**Comment by Mane76** [comment]
sure, but what is the sense behind ? 

Pure sending a beacon to RF or APRS-IS would work, but the way back (if someone wants to drop you a message would be broken up. 
For this case you would need to connect to APRS-IS via YAAC and use the iGate as pure hardware lora module. All "intelligence" would have to be done by YAAC. There are more elegant solutions.

Maybe you explain what you are targeting, then we can help you to find a suitable solution.

**Comment by sysadminpower2019** [comment]
so if i am supposed to data transfered when using socat, I don't see anything. 
So there is still some issue

**Comment by sysadminpower2019** [comment]
![Screenshot from 2024-07-19 18-20-31](https://github.com/user-attachments/assets/bdf04c1c-02ff-4a9f-bcfd-ce622a13f911)

**Comment by Mane76** [comment]
> so if i am supposed to data transfered when using socat, I don't see anything. So there is still some issue

How do you try to transfer data ?

**Comment by sysadminpower2019** [comment]
I am not sure what you mean, which data exactly? isn't this simply supposed to show any data sent from YAAC when using the tnc option?

**Comment by wb2osz** [comment]
I tried using YAAC with the LoRa Digi/IGate and it worked fine with TCP KISS.
Of course, you need to "Select APRS Server" in the LoRa Digi/IGate configuration.

Here is another approach to provide a bridge between LoRa APRS and the existing regular APRS RF network.
[https://github.com/wb2osz/direwolf-doc/blob/main/APRS-LoRa-VHF-APRS-Bridge.pdf ](https://github.com/wb2osz/direwolf-doc/blob/main/APRS-LoRa-VHF-APRS-Bridge.pdf )

73,
John WB2OSZ

**Comment by sysadminpower2019** [comment]
then something may be wrong with my network @wb2osz becuase nothing seems to work as expected. 
the kissutil does seem to connect but i don't know much more after that.

**Comment by Mane76** [comment]
> so if i am supposed to data transfered when using socat, I don't see anything. So there is still some issue

could you try to start socat with -v option ? Then it should show you more

**Comment by sysadminpower2019** [comment]
nothing at all.
I even enabled the syslog option and there is only one entry 
APRSIS TX / StartUp_Status / https://github.com/richonguzman/LoRa_APRS_iGate 2024.06.15 
nothing more is being written to the logs. 
Sending and receiving packets to and from the module itself is fine and the module and it's SSID can be found on aprs-fi.

**Comment by sysadminpower2019** [comment]
ah finally some data appeared on the socat -v output, although the syslog is still just one entry

**Comment by Mane76** [comment]
ok, syslog is not working when on TNC mode, so this is normal.
When you RX or TX data it could be read in clear-text in the 1st digit of every line...

**Comment by sysadminpower2019** [comment]
in any case the data that appears on the syslog output is only what is being sent from YAAC, nothing else. So the data is still not reaching the lora module for some reason.

**Comment by Mane76** [comment]
basic question, in iGate config, have you:

- Section "Station", Symbol, set to "Red Diamond with L", entered valid callsign
- Section "APRS-IS", enable "Enable APRS-IS connection" and "Gate APRS-IS Messages to RF"
- Section "Beaconing", enable "Send our beacon to APRS-IS"
- Section "LoRa", enable "Enable LoRa-TX" and "Enable LoRa-RX" and set to same freq 

?

**Comment by sysadminpower2019** [comment]
Yes, this is why the iGate works and send packages to aprs.fi without an issue. There is something else happening when trying to connect to the igate either via serial or tcp/ip which doesn't seem to work

**Comment by Mane76** [comment]
what is the callsign and ssid which is on iGate and YAAC ?

**Comment by sysadminpower2019** [comment]
i'd rather not provide my callsign, in any case the callsign with the iGate ssid is clearly visible on aprs.fi.

**Comment by Mane76** [comment]
please have a look on https://www.findu.com/cgi-bin/raw.cgi?call=CALLSIGN and replace your callsign and ssid in this link. 

The link is RAW APRS_IS data.
Check if something was TX from the iGate which looks like what you wanted to have.

**Comment by sysadminpower2019** [comment]
its interesting because now of all of a sudden the two ssid's from YAAC just appeared on APRS.fi. So I really don't know what is going on
